### PR TITLE
Add Emotion Detection Node

### PR DESCRIPTION
**Pull request by derkmed**
This PR should be stacked on https://github.com/PTG-Kitware/angel_system/pull/296.  Run ` tmuxinator start eval_vocalized_emotional_detection`  to demo.

### CODE REVIEW

File: ros/angel_system_nodes/angel_system_nodes/base_emotion_detector.py [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+                msg.utterance_text, classification, confidence_score
+            )
+
+    def publish_detected_emotion(
+        self, utterance: str, classification: str, confidence_score: float
+    ):
+        """
+        Handles message publishing for an utterance with a detected emotion classification.
+        """
+        emotion_msg = InterpretedAudioUserEmotion()
```
Comment (joshanderson-kw) [inline ros/angel_system_nodes/angel_system_nodes/base_emotion_detector.py:158]: Can you also fill out the `header` field here? Could fill in the timestamp and the frame_id fields

File: tmux/eval_vocalized_emotional_detection.yml [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+           -p interp_user_intent_topic:=interp_user_intent_topic
+  - emotion_detection:
+      layout: even-vertical
+      panes:
+        - base_emotion_detection: ros2 run angel_system_nodes base_emotion_detector --ros-args
+           -r __ns:=${ROS_NAMESPACE}
+           -p expect_user_intent_topic:=expect_user_intent_topic
+           -p interp_user_intent_topic:=interp_user_intent_topic
+           -p user_emotion_topic:=base_emotion_topic
+        - gpt_emotion_detection: ros2 run angel_system_nodes gpt_emotion_detector --ros-args
```
Comment (joshanderson-kw) [inline tmux/eval_vocalized_emotional_detection.yml:105]: Similar question here to the previous PR, do both of these nodes need to run or just one or the other?

File: ros/angel_msgs/msg/InterpretedAudioUserEmotion.msg [hunk 3]
```diff
@@ -0,0 +1,22 @@
+#
+# Message communicating an interpreted User Audio from an audio-based user
```
[the remark this answers is not in the document]
Comment (derkmed) [reply ros/angel_msgs/msg/InterpretedAudioUserEmotion.msg:2]: Done.

File: ros/angel_system_nodes/angel_system_nodes/base_emotion_detector.py [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+    def __init__(self):
+        super().__init__(self.__class__.__name__)
+        self.log = self.get_logger()
+
+        parameter_names = [
+            IN_EXPECT_USER_INTENT_TOPIC,
+            IN_INTERP_USER_INTENT_TOPIC,
+            OUT_INTERP_USER_EMOTION_TOPIC,
+        ]
+        set_parameters = self.declare_parameters(
```
[the remark this answers is not in the document]
Comment (derkmed) [reply ros/angel_system_nodes/angel_system_nodes/base_emotion_detector.py:39]: Done.  Will make  sure to do this in Q&A once  this PR gets merged  too.

File: ros/angel_system_nodes/angel_system_nodes/base_emotion_detector.py [hunk 5]
```diff
@@ -0,0 +1,192 @@
+import queue
+import rclpy
+from rclpy.node import Node
+from termcolor import colored
+import threading
+from vaderSentiment.vaderSentiment import SentimentIntensityAnalyzer
```
[the remark this answers is not in the document]
Comment (derkmed) [reply ros/angel_system_nodes/angel_system_nodes/base_emotion_detector.py:6]: Good point. I forgot to add this dependency. I'll do that here.

File: ros/angel_system_nodes/angel_system_nodes/base_emotion_detector.py [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+IN_EXPECT_USER_INTENT_TOPIC = "expect_user_intent_topic"
+IN_INTERP_USER_INTENT_TOPIC = "interp_user_intent_topic"
+OUT_INTERP_USER_EMOTION_TOPIC = "user_emotion_topic"
+
+# Currently supported emotions. This is tied with the emotions
+# output to VaderSentiment (https://github.com/cjhutto/vaderSentiment) and
+# will be subject to change in future iterations.
+LABEL_MAPPINGS = {"pos": "positive", "neg": "negative", "neu": "neutral"}
+
+VADER_NEGATIVE_COMPOUND_THRESHOLD = -0.05
```
[the remark this answers is not in the document]
Comment (derkmed) [reply ros/angel_system_nodes/angel_system_nodes/base_emotion_detector.py:20]: Nope. These are  the advised vader thresholds. I'll add a link to the  documentation in the comments.

[the remark this answers is not in the document]
Comment (derkmed) [reply ros/angel_system_nodes/angel_system_nodes/base_emotion_detector.py:20]: Please see https://github.com/cjhutto/vaderSentiment#about-the-scoring

File: ros/angel_system_nodes/angel_system_nodes/base_emotion_detector.py (hunk 1 above)
Comment (derkmed) [reply ros/angel_system_nodes/angel_system_nodes/base_emotion_detector.py:158]: Oh  yes. Similarly to  PR#296. Will  do  that.

File: ros/angel_system_nodes/angel_system_nodes/base_emotion_detector.py [hunk 7]
```diff
@@ -0,0 +1,192 @@
...
+        return (classification, confidence)
+
+    def get_inference(self, msg):
+        """
+        Abstract away the different model inference calls depending on the
+        node's configure model mode.
+        """
+        return self._get_vader_sentiment_analysis(msg.utterance_text)
+
+    def listener_callback(self, msg):
```
[the remark this answers is not in the document]
Comment (derkmed) [reply ros/angel_system_nodes/angel_system_nodes/base_emotion_detector.py:130]: Renamed to `intent_detection_callback(...)`.

File: tmux/eval_vocalized_emotional_detection.yml (hunk 2 above)
Comment (derkmed) [reply tmux/eval_vocalized_emotional_detection.yml:105]: Same answer. Only one of them needs to run for emotion detection. This .yml runs both just for testing purposes.

Additionally, I will relocate these .ymls  to their own .yml folder in a later  PR.

**Review by joshanderson-kw — APPROVED** [review]
LGTM. Thanks, @derkmed!

**Pull request merged**